testcases/01_get_request.py

```diff
@@ -0,0 +1,188 @@
+    if stderr:
+        print(f'{cmdline} =========================== fail')
+        print(stderr)
+
+    cmdline = f"cleos get currency balance eosio.token {newaccount}"
+    process = Popen(cmdline, stdout=PIPE, stderr=PIPE,shell=True)
+    stdout, stderr = process.communicate()
+    if stdout:
+        account2_balance2 = stdout
+        print("account2_balance2 ==",account2_balance2)
+    if stderr:
+        print(f'{cmdline} =========================== fail')
+        print(stderr)
+    #判断是否与预期相等
+    if (float(account1_balance1[:-4])- float(account1_balance2[:-4])) == (float(account2_balance2[:-4]) - float(account2_balance1[:-4])):
+        print(f'{cmdline2} ===========================  ok')
+    else:
+        print(f'{cmdline2} ===========================  fail')
+
+#system contract
+def system_contract(params):
+    #reprod
+    newaccount= params["newaccount"]
+    bppubkey = params["bppubkey"]
+    cmdline = f"cleos system regproducer {newaccount} {bppubkey} https://www.xxx.com 900"
+    process = Popen(cmdline, stdout=PIPE, stderr=PIPE,shell=True)
+    time.sleep(2)
+    cmdline1 = "cleos get table eosio eosio producers -l 100"
+    process = Popen(cmdline1, stdout=PIPE, stderr=PIPE,shell=True)
+    stdout, stderr = process.communicate()
+    if stdout:
+        result = json.loads(stdout)
+        pro = result['rows']
+        for i in pro:
+             if i['owner'] == newaccount:
+                 print(f'{cmdline} =========================== ok')
+                 res = True
+    if stderr:
+        print(f'{cmdline} =========================== fail')
+        print(stderr)
+        res = False 
+    
+    cmdline = f"cleos system unregprod {newaccount}"
+    process = Popen(cmdline, stdout=PIPE, stderr=PIPE,shell=True)
+    time.sleep(2)
+    cmdline1 = "cleos get table eosio eosio producers -l 100"
+    process = Popen(cmdline1, stdout=PIPE, stderr=PIPE,shell=True)
+    stdout, stderr = process.communicate()
+    if stdout:
+        result = json.loads(stdout)
+        pro = result['rows']
+        for i in pro:
+            if i['owner'] == newaccount:
+                print(f'{cmdline} =========================== ok')
+                res = True
+    if stderr:
+        print(f'{cmdline} =========================== fail')
+        print(stderr) 
+        res = False 
+    return res    
+
+def main():
+    if len(sys.argv)<1:
+        print('ERROR:Please supply the param file')
+        return False
+    param_dict, param_str = None, ''
+    with open(sys.argv[1], 'r') as fp:
+        param_str = fp.read()
+        param_dict = json.loads(param_str)
+    if not param_dict:
+        print('ERROR:param file can NOT be empty')
+        return False
+    print('Get params:', sys.argv[0], param_str)
+    res = get_info(param_dict["chain_id"])
+    if not res:
+        return False
+    system_contract(param_dict)
+    transfer(param_dict)
+    return True
+
+if __name__ == '__main__':
+    sys.exit( 0 if main() else 1)
```

Comment:
after you have finished your code, please run `python3 startmonster.py` to check the script works well

Reply:
i used python3.6，because httpapi/  used python3.6